File: python-single/src/gql/meters/meterBlock.py
from gql.common.enums.modelMutation import ModelMutation


class MeterBlock:
  def __init__(
    self,
    mutation,
    index,
    row,
    column,
    width,
    height,
    layout,
    options,
    meters,
    immersiveMeters,
    dynamics1Meters,
    dynamics2Meters,
    loudnessMeters,
    downmixMeters,
    surroundDownmixMeters,
    phaseMeters,
    videoMeters,
    meterLabels,
    colour,
    labels,
    path,
    isFader,
    autoFader,
    channelIndex,
    layer,
    format,
    loudnessFormat,
    isCurrentAssignedFader,
  ):
    self.mutation = mutation
    self.index = index
    self.row = row
    self.column = column
    self.width = width
    self.height = height
    self.layout = layout
    self.options = options
    self.meters = meters
    self.immersiveMeters = immersiveMeters
    self.dynamics1Meters = dynamics1Meters
    self.dynamics2Meters = dynamics2Meters
    self.loudnessMeters = loudnessMeters
    self.downmixMeters = downmixMeters
    self.surroundDownmixMeters = surroundDownmixMeters
    self.phaseMeters = phaseMeters
    self.videoMeters = videoMeters
    self.meterLabels = meterLabels
    self.colour = colour
    self.labels = labels
    self.path = path
    self.isFader = isFader
    self.autoFader = autoFader
    self.channelIndex = channelIndex
    self.layer = layer
    self.format = format
    self.loudnessFormat = loudnessFormat
    self.isCurrentAssignedFader = isCurrentAssignedFader

  def set_mutation(self, mutation: ModelMutation):
    self.mutation = mutation

File: rust-single/src/gql/meters/meterBlock.rs
#![allow(non_snake_case)]

use crate::{klv::common::{enums::{format::Format, meterLabelBackgroundColour::MeterLabelBackgroundColour, meterLayout::MeterLayout}, meterOptions::MeterOptions}, gql::common::enums::modelMutation::ModelMutation};

use super::{meterBlockLabels::MeterBlockLabels, meterBar::MeterBar, videoMeter::VideoMeter, meterLabel::MeterLabel};


#[derive(Debug, Clone)]
pub struct MeterBlock {
  pub mutation: ModelMutation,
  pub index: i32,
  pub row: i32,
  pub column: i32,
  pub width: i32,
  pub height: i32,
  pub layout: MeterLayout,
  pub options: MeterOptions,
  pub meters: Vec<MeterBar>,
  pub immersiveMeters: Vec<MeterBar>,
  pub dynamics1Meters: Vec<MeterBar>,
  pub dynamics2Meters: Vec<MeterBar>,
  pub loudnessMeters: Vec<MeterBar>,
  pub downmixMeters: Vec<MeterBar>,
  pub surroundDownmixMeters: Vec<MeterBar>,
  pub phaseMeters: Vec<MeterBar>,
  pub videoMeters: Vec<VideoMeter>,
  pub meterLabels: Vec<MeterLabel>,
  pub colour: MeterLabelBackgroundColour,
  pub labels: MeterBlockLabels,
  pub path: String,
  pub isFader: bool,
  pub autoFader: Option<i32>,
  pub channelIndex: i32,
  pub layer: String,
  pub format: Format,
  pub loudnessFormat: Format,
  pub isCurrentAssignedFader: bool,
}

impl MeterBlock {
  pub fn set_mutation(&mut self, mutation: ModelMutation) {
    self.mutation = mutation;
  }
}
